ledger/src/blockstore_db.rs

```diff
@@ -1991,28 +1991,34 @@ fn process_cf_options_advanced<C: 'static + Column + ColumnName>(
     }
 }
 
-fn get_db_options(access_type: &AccessType) -> Options {
+fn get_db_options(blockstore_options: &BlockstoreOptions) -> Options {
     let mut options = Options::default();
 
     // Create missing items to support a clean start
     options.create_if_missing(true);
     options.create_missing_column_families(true);
 
-    // Per the docs, a good value for this is the number of cores on the machine
-    options.increase_parallelism(num_cpus::get() as i32);
-
+    // rocksdb builds two threadpools: low and high priority. The low priority
+    // pool is used for compactions whereas the high priority pool is used for
+    // memtable flushes. Separate pools are created so that compactions are
+    // unable to stall memtable flushes (which could stall memtable writes).
     let mut env = rocksdb::Env::new().unwrap();
-    // While a compaction is ongoing, all the background threads
-    // could be used by the compaction. This can stall writes which
-    // need to flush the memtable. Add some high-priority background threads
-    // which can service these writes.
-    env.set_high_priority_background_threads(4);
+    env.set_low_priority_background_threads(
+        blockstore_options.num_rocksdb_compaction_threads.get() as i32,
+    );
+    env.set_high_priority_background_threads(
+        blockstore_options.num_rocksdb_flush_threads.get() as i32
+    );
     options.set_env(&env);
+    // The value set in max_background_jobs can grow but not shrink threadpool
+    // sizes. So, set this value to 2 (the default value and 1 low / 1 high) to
+    // avoid rocksdb from messing with the sizes we previously configured.
+    options.set_max_background_jobs(2);
```

Comment:
this is slightly confusing to me. Might be nice to either:
1. Change this to something like `options.set_max_background_jobs(blockstore_options.num_rocksdb_compaction_threads + blockstore_options.num_rocksdb_flush_threads);`
2. Specify in the comment that the low/high priority thread counts will override the value that is set here as long as `max_background_jobs` is lower

Reply:
I thought about doing 1, but this don't work if our number differ from a 3 low:1 high ratio. I linked the snippets under "Additional Context for setting threadpool sizes directly" that show why this is the case.

As for 2, any specific recommendations on the comment ?
```
    // The value set in max_background_jobs can grow but not shrink threadpool
    // sizes. So, set this value to 2 (the default value and 1 low / 1 high) to
    // avoid rocksdb from messing with the sizes we previously configured.
```
Maybe changing the "from messing with the sizes we previously configured" comment to be more explicit about what it is referring to would do the trick ?

Reply:
Reworded in https://github.com/anza-xyz/agave/pull/4214/commits/09bf071068d5ff43fa545f053e58875e550098a9.

Technically, 2 is the default for this value if we do not call `set_max_background_jobs()`, but given that this couples into thread pool configuration, I thought it was friendlier to the next person reading it to set the value explicitly and leave a comment instead of forcing them to go dig through rocksd too